Add prompt for variable creation

paulkaplan opened the pull request:
A simple first iteration of a new variable prompt. I wanted to get some feedback on the way this is implemented @rschamp. (The css definitely needs some work, and I avoided the header bar from the mockup because there isn't the data to fill it). 
_Update:_ Added the prompt header and refactored container/component divide.
<img width="845" alt="screen shot 2017-04-26 at 10 17 26 am" src="https://cloud.githubusercontent.com/assets/654102/25439124/94c0d13e-2a69-11e7-8850-1dd5eb2f3101.png">

Fixes https://github.com/LLK/scratch-blocks/issues/535

rschamp commented on src/components/prompt/prompt.jsx:
```diff
@@ -0,0 +1,74 @@
+    placeholder: '...'
+};
+
+module.exports = PromptComponent;
```
This should be split into a container and component.

paulkaplan replied:
How do you mean? What part should be a container?

rschamp replied:
The component should only call props for its handlers, and the container should implement them.

It doesn't matter too much at this point, other than for consistency, but as the components grow, it provides the split between the presentation and business logic.

paulkaplan commented:
@rschamp thanks for the feedback. I separated the container/component boundary and updated the binding with scratch-blocks. I also added a header per the design spec. Should be ready for a real review!
